### PR TITLE
add class lib templates for winforms

### LINKED ISSUE #1414: Support Class Library project templates
This is to bring us to experience parity with desktop for WPF on .NET Core 3.0.

On desktop today you can add a class library project to your existing Winforms solution and everything will just work. Due to complexities in how Framework References work this will not light up automatically. Instead we will need a Winforms Class library template that has the correct framework references.

### PR DESCRIPTION
**Pull request by jmarolf**
fixes #1414

### CODE REVIEW

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-CSharp/Company.ClassLibrary1.csproj [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+  <PropertyGroup>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' == ''">netcoreapp3.0</TargetFramework>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride</TargetFramework>
+    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.WinFormsApplication1</RootNamespace>
+    <LangVersion Condition="'$(langVersion)' != ''">$(ProjectLanguageVersion)</LangVersion>
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+</Project>
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-CSharp/Company.ClassLibrary1.csproj]: Could you please fix the line ending by opening this file in a VS text editor?

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Import Include="System.Data" />
+    <Import Include="System.Drawing" />
+    <Import Include="System.Windows.Forms" />
+  </ItemGroup>
+
+</Project>
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj]: Could you please fix the lie ending?

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Class1.vb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+﻿Public Class Class1
+
+End Class
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Class1.vb]: one more :)

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+
+  <PropertyGroup>
+    <RootNamespace>Company.WinFormsApplication1</RootNamespace>
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj]: Looks like a copy/paste error. This should be ClassLibrary1 :)

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-CSharp/Company.ClassLibrary1.csproj [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">
+
+  <PropertyGroup>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' == ''">netcoreapp3.0</TargetFramework>
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride</TargetFramework>
+    <RootNamespace Condition="'$(name)' != '$(name{-VALUE-FORMS-}safe_namespace)'">Company.WinFormsApplication1</RootNamespace>
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-CSharp/Company.ClassLibrary1.csproj]: Shuoldn't rootnamespace name say `ClassLibrary`

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    <TargetFramework Condition="'$(TargetFrameworkOverride)' != ''">TargetFrameworkOverride</TargetFramework>
+    <LangVersion Condition="'$(langVersion)' != ''">$(ProjectLanguageVersion)</LangVersion>
+    <UseWindowsForms>true</UseWindowsForms>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Import Include="System.Data" />
+    <Import Include="System.Drawing" />
+    <Import Include="System.Windows.Forms" />
+  </ItemGroup>
```
Comment (Tanya-Solyanik) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj]: Do you need this item group in addition to the `UseWinforms` property set to true?

**Comment by Tanya-Solyanik** [comment]
@jmarolf  - what is the urgency of this work and had it been reviewed by Roslyn team? I don't see how a class library with no controls is related to winforms. We used to have a class library and a `winforms user control library`.
image

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj (hunk 6 above)
Comment (jmarolf) [reply pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/Company.ClassLibrary1.vbproj]: Winforms developers on VB expect the default namespaces to be auto-imported for them

**Comment by jmarolf** [comment]
@Tanya-Solyanik 	this is a different scenario.  Users used to be able to add a class library and reference it from a Winforms project. Adding a normal Core class library won't show the tools it add winforms item templates in core

File: pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/.template.config/template.json [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      "description": "Restore NuGet packages required by this project.",
+      "manualInstructions": [
+        { "text": "Run 'dotnet restore'" }
+      ],
+      "actionId": "210D431B-A78B-4D2F-B762-4ED3E3EA9025",
+      "continueOnError": true
+    },
+    {
+      "condition": "(HostIdentifier != \"dotnetcli\" && HostIdentifier != \"dotnetcli-preview\")",
+      "description": "Opens Class.vb in the editor",
```
Comment (RussKie) [inline pkg/Microsoft.Dotnet.WinForms.ProjectTemplates/content/WinFormsLibrary-VisualBasic/.template.config/template.json]: Should this be the following?
```suggestion
      "description": "Opens Class1.vb in the editor",
```

**Comment by Tanya-Solyanik** [comment]
>Users used to be able to add a class library and reference it from a Winforms project. Adding a normal Core class library won't show the tools it add winforms item templates in core

Thank you for the explanation!

**Comment by OliaG** [comment]
Looks good!

**Pull request merged**